lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4281,19 +4283,17 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
   // i8* <ptr> is a pointer on a GV, which can carry optinal variadic
   // clang::annotation attribute expression arguments.
   case Intrinsic::ptr_annotation: {
-    // Strip all bitcast and addrspace casts from the pointer argument:
-    //   llvm annotation intrinsic only takes i8*, so the original pointer
-    //   probably had to loose its addrspace and its original type.
-    Value *AnnotSubj = II->getArgOperand(0);
-    while (isa<BitCastInst>(AnnotSubj) || isa<AddrSpaceCastInst>(AnnotSubj)) {
-      AnnotSubj = cast<CastInst>(AnnotSubj)->getOperand(0);
+    Value *AnnotSubj = nullptr;
+    if (auto *BI = dyn_cast<BitCastInst>(II->getArgOperand(0))) {
+      AnnotSubj = BI->getOperand(0);
+    } else {
```

Comment:
Do we even need this code with this approach?

Reply:
Shall we then also skip addrspace casts?

Reply:
If we don't skip the bitcasts - then in the reverse translation the `OpDecorate` will get translated to `llvm.ptr.annotation` in even more cases, as the heuristic (discussed) above will not trigger. I think it is important to keep as many `llvm.var.annotations` as possible, as they provide stronger guarantees in the LLVM IR than `llvm.ptr.annotations`, and are more likely to get interpenetrated by the backendm rather than dropped.